test/sqllogictest/interval.slt

```diff
@@ -127,6 +127,76 @@ SELECT INTERVAL '1-2 3 4:5:6.7' DAY TO HOUR;
 ----
 3 days 04:00:00
 
+query T
+SELECT INTERVAL '12:34';
+----
+12:34:00
+
+query T
+SELECT INTERVAL '12:34' HOUR;
+----
+12:00:00
+
+query T
+SELECT INTERVAL '12:34' MINUTE;
+----
+12:34:00
+
+query T
+SELECT INTERVAL '12:34' SECOND;
+----
+12:34:00
+
+query T
+SELECT INTERVAL '12:34' HOUR TO MINUTE;
+----
+12:34:00
+
+query T
+SELECT INTERVAL '12:34' HOUR TO SECOND;
+----
+12:34:00
+
+query T
+SELECT INTERVAL '12:34' MINUTE TO SECOND;
+----
+00:12:34
```

Comment:
Would you mind throwing these tests in here, as well?

```
materialize=> SELECT INTERVAL '12:34' MONTH TO SECOND;
 interval 
----------
 12:34:00
(1 row)

materialize=> SELECT INTERVAL '12:34' DAY TO SECOND;
 interval 
----------
 12:34:00
(1 row)
```

my thinking here is ensuring that we don't somehow misconstrue the head here as specifying that the leading element is anything greater than an hour